### PR TITLE
Update fetch & save/send Draft policies

### PR DESCRIPTION
**Pull request by KevinBoulongne**
- Save Draft when leaving NewMessage screen
- Fetch Draft when opening NewMessage screen
- Update Message when Draft is fetched

Depends on #321

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt [hunk 1]
```diff
@@ -72,5 +72,10 @@ object MessageController {
         }
         realm?.let(block) ?: RealmDatabase.mailboxContent().writeBlocking(block)
     }
+
+    fun updateMessage(uid: String, realm: MutableRealm? = null, onUpdate: (message: Message) -> Unit) {
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt:76]: It is only called by `fetchDraft`.
And the realm is never null. You should make it an extension of MutableRealm like `fetchDraft`.

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt [hunk 2]
```diff
@@ -129,27 +125,18 @@ class NewMessageViewModel : ViewModel() {
...
-
-                val draft = DraftController.getDraft(currentDraftLocalUuid!!, this) ?: return@writeBlocking
-                DraftController.executeDraftAction(draft, mailboxUuid, this)
-            }
+            saveDraftToLocal(currentDraftLocalUuid!!, DraftAction.SAVE)
         }
     }
 
     fun saveToLocalAndFinish(action: DraftAction) = viewModelScope.launch(Dispatchers.IO) {
         val draftLocalUuid = currentDraftLocalUuid ?: return@launch
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt:133]: Can `currentDraftLocalUuid` really be null?

**Review by JorisBodin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt:133]: No it cannot.

It's fixed.

**Review by sirambd — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt [hunk 3]
```diff
@@ -129,27 +125,17 @@ class NewMessageViewModel : ViewModel() {
...
-            saveDraftToLocal(draftLocalUuid, action)
-        }
+        saveDraftToLocal(action)
         shouldCloseActivity.postValue(true)
     }
 
-    private fun MutableRealm.saveDraftToLocal(draftLocalUuid: String, action: DraftAction) {
-        DraftController.updateDraft(draftLocalUuid, this) { draft ->
+    private fun saveDraftToLocal(action: DraftAction) {
+        DraftController.updateDraft(currentDraftLocalUuid!!) { draft ->
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt:138]: if it cannot be null, then remove the null

**Review by sirambd — CHANGES_REQUESTED** [review]

**Review by JorisBodin — APPROVED** [review]

**Review by sirambd — APPROVED** [review]

**Pull request merged**